dts/bindings/w1/adi,max32-w1.yaml

```diff
@@ -0,0 +1,64 @@
+# Copyright (c) 2023-2024 Analog Devices, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+description: ADI MAX32xxx MCUs 1-Wire Master
+
+include: [w1-master.yaml, pinctrl-device.yaml]
+
+compatible: "adi,max32-w1"
+
+properties:
+  reg:
+    required: true
+
+  clocks:
+    required: true
+
+  interrupts:
+    required: true
+
+  pinctrl-0:
+    required: true
+
+  pinctrl-names:
+    required: true
+
+  internal-pullup:
+    required: true
+    type: int
+    enum: [0, 1]
+    description: |
+      Set this field to enable the internal pullup resistor.
+      0 - Internal pullup disabled.
+      1 - Internal pullup enabled.
+
+  external-pullup:
+    required: true
+    type: int
+    enum: [0, 1, 2]
+    description: |
+      Set this field to enable the external pullup.
+      0 - Pullup pin is active high when enabled.
+      1 - Pullup pin is active low when enabled.
```

Comment:
Which configuration bits do you want to set with that property. And is available on all models?